grafana/grafana
# Loki: Filter by labels based on the type of label (structured, indexed, parsed)

**What is this feature?**

This feature improves the filter actions in the LogDetails for Loki. When the dataframe contains a definition about the label type, indexed labels will always be added to the stream selector. Parsed labels and keys from structured metadata will be added as LabelFilter expressions.

https://github.com/grafana/grafana/assets/8092184/1ff868c4-e6a4-46ba-bdf0-c003dc916bbe

**How to test**

1. Without the `lokiStructuredMetadata` feature toggle nothing should change.
2. With `lokiStructuredMetadata` set, labels should be added accordingly.
